doc/developer-guides/hld/hld-devicemodel.rst

```diff
@@ -789,6 +789,137 @@ the bus hierarchy would be:
+VM. When the user enables PTM, the passthru device is connected to a virtual
+root port instead of the host bridge.  Here is an example launch script that
+enables PTM on a passthru Ethernet card:
+
+.. code-block:: bash
+   :emphasize-lines: 5
+
+   acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+      -s 3,virtio-blk,uos-test.img \
+      -s 4,virtio-net,tap0 \
+      -s 5,virtio-console,@stdio:stdio_port \
+      -s 6,passthru,a9/00/0,enable_ptm \
+      --ovmf /usr/share/acrn/bios/OVMF.fd
+
+And here is the bus hierarchy from guest (as shown by the ``lspci`` command)::
+
+   sudo lspci -tv
+   -[0000:00]-+-00.0  Network Appliance Corporation Device 1275
+              +-03.0  Red Hat, Inc. Virtio block device
+              +-04.0  Red Hat, Inc. Virtio network device
+              +-05.0  Red Hat, Inc. Virtio console
+              \-06.0-[01]----00.0  Intel Corporation Device 15f2
+
+
+PTM Implementation Notes
+------------------------
+
+To simplify PTM support implementation, the virtual root port only supports the
+most basic PCIe configuration and operation, in addition to PTM capabilities.
+
+In Guest VM post-launched scenarios, you enable PTM by setting the
+``enable_ptm`` option for the pass through device (as shown above).
+
+.. figure:: images/PTM-hld-PTM-flow.png
+   :align: center
+   :width: 700
+   :name: ptm-flow
+
+   PTM-enabling workflow in post-launched VM
+
+As shown in :numref:`ptm-flow`, PTM is enabled in the root port during the
+hypervisor startup. Device model then checks whether the pass through device
+supports PTM requestor capabilities and whether the corresponding root port
+supports PTM root capabilities, as well as some other sanity checks.  If an
+error is detected during these checks, the error will be reported and ACRN will
+not enable PTM on guest. This doesn’t prevent the user from launching the Guest
+VM and passing through the device to the Guest VM.  If no error is detected,
+device model will use ``add_vdev`` hypercall to add a virtual root port (VRP),
+acting as the PTM root, to Guest before passing through the device to Guest.
+
+.. figure:: images/PTM-hld-PTM-passthru.png
+   :align: center
+   :width: 700
+   :name: ptm-vrp
+
+   PTM-enabled PCI device pass-through to post-launched VM
+
+:numref:`ptm-vrp` show that, after enabling PTM, the passthru device connects to
+the virtual root port instead of the virtual host bridge.
+
+To use PTM in a virtualized environment, you may want to first verify that PTM
+is supported by the device and is enabled on the bare metal machine.
+If supported, follow these steps to enable PTM on post-launched guest:
+
+1. Make sure that PTM is enabled in guest kernel.  In Linux kernel, for example,
+   set “CONFIG_PTM=on”.
+2. Not every PCI device supports PTM.  One example that does it the Intel I225-V
+   Ethernet controller.  If you passthru this card to guest,  make sure guest
+   uses a version of the IGC driver that supports PTM.
```

Comment:
What is this IGC driver? Can we include a pointer to it?

Reply:
I guess IGC here refers to the Linux kernel `CONFIG_IGC` driver?

It sounds like not all versions of the IGC drivers support PTM, can we indicate which does?